prof/util.py

```diff
@@ -0,0 +1,136 @@
+import collections
+import contextlib
+import functools
+
+
+def histogram(stream):
+    counts = collections.defaultdict(int)
```

Comment:
Is `histogram` here just reimplementing `collections.Counter`? Although I usually don't use it this way it appears it takes an iterable and I would guess does exactly what histogram is doing here.

https://docs.python.org/3/library/collections.html#collections.Counter

Reply:
Cool, thanks – I didn't see this added in Python 3. (It's not hard to do, but it did seem like something that belonged there.)